packages/vats/src/core/startWalletFactory.js

```diff
@@ -223,32 +225,60 @@ export const startWalletFactory = async (
     privateArgs: {
       storageNode: walletStorageNode,
       walletBridgeManager,
-      walletReviver: E(ppFacets.creatorFacet).getWalletReviver(),
+      walletReviver: E(E.get(ppFacetsP).creatorFacet).getWalletReviver(),
```

Comment:
we pass a promise in `privateArgs`? I wonder how that works with upgrade... Do settled promises stay settled across upgrades?

Reply:
When a remote promise settles, the kernel is informed, and any vat that has subscribed to that promise will get the resulting data (or presence).  At that point, it is safe for either vat to upgrade, without the settled promise somehow interfering.